src/main/java/org/wildfly/security/auth/realm/FileSystemSecurityRealm.java

```diff
@@ -574,6 +592,12 @@ private void writeIdentity(final XMLStreamWriter streamWriter, final LoadedIdent
             streamWriter.writeEndDocument();
         }
 
+        public void dispose() {
+            // Release the write lock for this realm identity
+            final ReentrantReadWriteLock realmIdentityLock = getRealmIdentityLockForName(name);
+            realmIdentityLock.writeLock().unlock();
```

Comment:
Another note about that.  Ownership is a critical concept for locks; specifically, preventing someone else from releasing your lock is of great importance.  If you do opt to take advantage of ReadWriteLock, I would advise that readLock/writeLock each construct a new instance of Lock which represents a distinct ownership, such that unlock() on one does not release the lock() on another.  If you do not use the ReadWriteLock/Lock interfaces, then single lockExclusive and lockShared methods which return a handle that can be released (a subtype of AutoCloseable perhaps) would be a good option.